### PR TITLE
Refactoring by call (fixes that have been discussed)

### PR DESCRIPTION
**Pull request by KonstantinVerenev**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,22 +1,5 @@
...
-              name: 'MainScreen'
-            }
-          }
-        ]
-      }
-    }
-  });
-});
\ No newline at end of file
+Navigation.events().registerAppLaunchedListener(rootNavigation);
```
Comment (AnnZar12) [inline index.js]: Please, add an empty line

File: navigation.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
+import { Navigation } from "react-native-navigation";
+
+import MainScreen from './src/screens/MainScreen';
+import DetailScreen from './src/screens/DetailScreen';
+
+import colors from "./src/constants/colors";
```
Comment (AnnZar12) [inline navigation.js]: Could you please add ESLint rules for quotes? (right now, there is a mix of single and double quotes)

File: src/components/CatsListItem.tsx [hunk 3]
```diff
@@ -1,94 +1,80 @@
...
-        />
-        <Text style={styles.itemDesc}>{props.item.description}</Text>
+        <Text style={styles.itemDesc}>{item.description}</Text>
       </TouchableOpacity>
     </View>
   );
 };
 
 const styles = StyleSheet.create({
   itemContainer: {
```
Comment (AnnZar12) [inline src/components/CatsListItem.tsx]: You could skip using the 'item' prefix in each style name. Just 'container', 'description', 'header' enough

File: src/components/CatsListItem.tsx [hunk 4]
```diff
@@ -1,94 +1,80 @@
...
-        }}>
+      <TouchableOpacity style={styles.cardContainer} onPress={onOpenDetailScreen}>
         <View style={styles.itemHeader}>
-          <Text style={{ ...styles.headerText, ...{ fontFamily: 'OpenSans-Bold' } }}>
-            {props.item.name}
-          </Text>
-          <Text style={{ ...styles.headerText, ...{ fontFamily: 'OpenSans-Regular' } }}>
-            {props.item.breed}
-          </Text>
+          <Text style={styles.headerText}>{item.name}</Text>
```
Comment (AnnZar12) [inline src/components/CatsListItem.tsx]: I would prefer object destructuring here to decrease `item` word usage

File: src/components/CatsListItem.tsx [hunk 5]
```diff
@@ -1,94 +1,80 @@
...
-    color: '#CED46A',
-    fontFamily: 'OpenSans-Regular',
-    fontSize: 16,
+  imageContainer: {
+    flex: 1,
+    alignItems: 'center',
   },
   catImage: {
     width: '90%',
     height: 200,
```
Comment (AnnZar12) [inline src/components/CatsListItem.tsx:88]: How about replacing strict values with `resizeMode`?

File: src/components/SearchInput.tsx [hunk 6]
```diff
@@ -1,34 +1,27 @@
...
-  container: {
-    height: '100%',
-  },
   inputField: {
-    height: '100%',
+    flex: 1,
     padding: 5,
-    textAlign: 'center',
     borderWidth: 1,
     borderColor: 'lightgrey',
```
Comment (AnnZar12) [inline src/components/SearchInput.tsx]: Please, move `lightgrey` color to the `colors`

File: src/screens/DetailScreen.tsx [hunk 7]
```diff
@@ -1,85 +1,75 @@
...
-        />
-        <Text style={styles.textItem}>Порода: {selectedCat?.breed}</Text>
-        <Text style={styles.textItem}>Возраст: {selectedCat?.age}</Text>
-        <Text style={styles.itemDesc}>{selectedCat?.description}</Text>
-      </LinearGradient>
-    </View>
-  );
-};
+    return (
+      <View style={styles.itemContainer}>
```
Comment (AnnZar12) [inline src/screens/DetailScreen.tsx]: Unnecessary wrapper. You could remove it

File: src/screens/DetailScreen.tsx [hunk 8]
```diff
@@ -1,85 +1,75 @@
...
-    background: {
-      color: '#07553B',
-    },
-    backButton: {
-      title: 'Назад',
-      color: '#CED46A',
-      fontFamily: 'OpenSans-Regular',
-    },
-  },
+  return <Text>Ошибка: CatId</Text>;
```
Comment (AnnZar12) [inline src/screens/DetailScreen.tsx]: Could you please add a more user-friendly message to the user?

File: src/screens/DetailScreen.tsx [hunk 9]
```diff
@@ -1,85 +1,75 @@
...
+          style={styles.linearGradient}>
+          <ScrollView contentContainerStyle={styles.scrollContainer}>
+            <Image
+              style={styles.catImage}
+              source={{
+                uri: img,
+              }}
+            />
+            <Text style={styles.textItem}>Порода: {breed}</Text>
+            <Text style={styles.textItem}>Возраст: {age}</Text>
```
Comment (AnnZar12) [inline src/screens/DetailScreen.tsx]: The same comment here: 'item' prefix is better to omit

File: src/screens/DetailScreen.tsx [hunk 10]
```diff
@@ -1,85 +1,75 @@
...
   catId: number;
 };
 
-const DetailScreen: NavigationFunctionComponent<DetailScreenProps> = (props) => {
-  const selectedCatId = props.catId;
+const DetailScreen: NavigationFunctionComponent<DetailScreenProps> = ({ catId }) => {
+  const selectedCat = Cats.find((cat) => cat.id === catId);
 
-  const selectedCat = Cats.find((cat) => cat.id === selectedCatId);
+  if (selectedCat) {
```
Comment (AnnZar12) [inline src/screens/DetailScreen.tsx]: I suggest a bit updating structure: 

```
if (!selectedCat){
   return <Text>${someErrorMesage}</Text>
}

const { img, breed, age, description } = selectedCat;

return <Component />
```

File: src/screens/DetailScreen.tsx [hunk 11]
```diff
@@ -1,85 +1,75 @@
...
+  },
+  scrollContainer: {
     alignItems: 'center',
-    width: '100%',
   },
   catImage: {
-    width: '95%',
-    height: 250,
+    width: '90%',
+    height: 230,
```
Comment (AnnZar12) [inline src/screens/DetailScreen.tsx:51]: The same comment here for resize mode

File: src/screens/MainScreen.tsx [hunk 12]
```diff
@@ -1,34 +1,57 @@
...
+
 import CatsList from '../components/CatsList';
 import SearchInput from '../components/SearchInput';
-import { Cats, CatType } from '../data/CatsData';
 
-const MainScreen: NavigationFunctionComponent = (props) => {
-  const [searchQuery, setSearchQuery] = useState('');
-  const [catsData, setCatsData] = useState<Array<CatType>>([]);
+const MainScreen: NavigationFunctionComponent = ({ componentId }) => {
+  // const [searchQuery, setSearchQuery] = useState('');
```
Comment (AnnZar12) [inline src/screens/MainScreen.tsx]: Could you please remove comments?

File: src/screens/MainScreen.tsx [hunk 13]
```diff
@@ -1,34 +1,57 @@
...
+    setCatsData(filtredCatsData);
+  };
+
+  const onOpenDetailScreen = useCallback(
+    (catId: number, catName: string) => {
+      Navigation.push(componentId, {
+        component: {
+          name: DETAIL_SCREEN,
+          passProps: {
+            catId: catId,
```
Comment (AnnZar12) [inline src/screens/MainScreen.tsx]: Just `catId` instead of `catId: catId`

File: src/components/CatsListItem.tsx [hunk 14]
```diff
@@ -1,94 +1,80 @@
...
   cardContainer: {
-    width: '95%',
+    flex: 1,
     marginTop: 20,
+    marginHorizontal: 10,
     paddingHorizontal: 10,
-    alignItems: 'center',
-    backgroundColor: '#07553B',
+    backgroundColor: colors.darkGreen,
     shadowColor: 'black',
```
Comment (AnnZar12) [inline src/components/CatsListItem.tsx]: Could you please move this black color to `colors` as well?

**Review by AnnZar12 — COMMENTED** [review]
Looks good!

Just a few things which you could look at:
1. Keep in mind and handle possible log text:

image
2. Check your code for comments and unused variables. You could add a ESLint rule for it
3.  Be careful with keyboardVerticalOffset
image

File: index.js (hunk 1 above)
Comment (KonstantinVerenev) [reply index.js]: Fixed

File: navigation.js (hunk 2 above)
Comment (KonstantinVerenev) [reply navigation.js]: fixed, and add rule

File: src/components/CatsListItem.tsx (hunk 3 above)
Comment (KonstantinVerenev) [reply src/components/CatsListItem.tsx]: fixed

File: src/components/CatsListItem.tsx (hunk 4 above)
Comment (KonstantinVerenev) [reply src/components/CatsListItem.tsx]: fixed

File: src/components/SearchInput.tsx (hunk 6 above)
Comment (KonstantinVerenev) [reply src/components/SearchInput.tsx]: done

File: src/screens/DetailScreen.tsx (hunk 7 above)
Comment (KonstantinVerenev) [reply src/screens/DetailScreen.tsx]: done

File: src/screens/DetailScreen.tsx (hunk 8 above)
Comment (KonstantinVerenev) [reply src/screens/DetailScreen.tsx]: done

File: src/screens/DetailScreen.tsx (hunk 9 above)
Comment (KonstantinVerenev) [reply src/screens/DetailScreen.tsx]: fixed

File: src/screens/DetailScreen.tsx (hunk 10 above)
Comment (KonstantinVerenev) [reply src/screens/DetailScreen.tsx]: Ok. Great idea

Comment (KonstantinVerenev) [reply src/screens/DetailScreen.tsx]: Fixed

File: src/screens/MainScreen.tsx (hunk 12 above)
Comment (KonstantinVerenev) [reply src/screens/MainScreen.tsx]: Fixed

File: src/screens/MainScreen.tsx (hunk 13 above)
Comment (KonstantinVerenev) [reply src/screens/MainScreen.tsx]: Fixed

File: src/components/CatsListItem.tsx (hunk 14 above)
Comment (KonstantinVerenev) [reply src/components/CatsListItem.tsx]: Moved

**Review by AnnZar12 — APPROVED** [review]

**Pull request merged**